### PR TITLE
Remove multiple overrides of /atom/New

### PR DESCRIPTION
**Pull request by coiax**
With the use of a temporary `SHOULD_NOT_OVERRIDE` in /atom/New(), a
number of unrequired overrides of New() were identified, and replaced
with Initialize() calls.

The order of before and after the parent call has been improved where it
was obvious that the order did not matter, with more esoteric ordering
remaining static to avoid breaking anything.

A number of STOP_PROCESSING macros were added where appropriate.

This commit does not replace all New() overrides, technically
INITIALIZE_IMMEDIATE overrides New(), and a number have distinct
comments warning against casual Initialize() replacements.

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Do you know why these were new? I'm not sure I like a sweeping change like this, it seems like we'll lose some information in the transition. On that topic: https://github.com/tgstation/tgstation/pull/31781#discussion_r144742943

File: code/_onclick/hud/plane_master_controller.dm [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	var/datum/hud/owner_hud
 
 ///Ensures that all the planes are correctly in the controlled_planes list.
-/atom/movable/plane_master_controller/New(hud)
+/atom/movable/plane_master_controller/Initialize(mapload, hud)
```
Comment (Cyberboss) [inline code/_onclick/hud/plane_master_controller.dm:9]: Remember the first argument of New (loc) is replaced with mapload

File: code/game/objects/effects/temporary_visuals/projectiles/projectile_effects.dm [hunk 2]
```diff
@@ -31,7 +31,7 @@
 	M.Turn(angle)
 	transform = M
 
-/obj/effect/projectile/New(angle_override, p_x, p_y, color_override, scaling = 1)
+/obj/effect/projectile/Initialize(mapload, angle_override, p_x, p_y, color_override, scaling = 1)
```
Comment (Cyberboss) [inline code/game/objects/effects/temporary_visuals/projectiles/projectile_effects.dm:34]: ^

File: code/game/objects/items/documents.dm [hunk 3]
```diff
@@ -37,8 +37,8 @@
 	var/forgedseal = 0
 	var/copy_type = null
 
-/obj/item/documents/photocopy/New(loc, obj/item/documents/copy=null)
-	..()
+/obj/item/documents/photocopy/Initialize(loc, obj/item/documents/copy=null)
```
Comment (Cyberboss) [inline code/game/objects/items/documents.dm:40]:
```suggestion
/obj/item/documents/photocopy/Initialize(mapload, obj/item/documents/copy=null)
```

File: code/game/objects/structures/petrified_statue.dm [hunk 4]
```diff
@@ -8,7 +8,7 @@
 	var/timer = 480 //eventually the person will be freed
 	var/mob/living/petrified_mob
 
-/obj/structure/statue/petrified/New(loc, mob/living/L, statue_timer)
+/obj/structure/statue/petrified/Initialize(mapload, loc, mob/living/L, statue_timer)
```
Comment (Cyberboss) [inline code/game/objects/structures/petrified_statue.dm:11]: loc isn't a param

File: code/game/objects/structures/transit_tubes/transit_tube.dm [hunk 5]
```diff
@@ -13,8 +13,8 @@
 	var/exit_delay = 1
 	var/enter_delay = 0
 
-/obj/structure/transit_tube/New(loc, newdirection)
-	..(loc)
+/obj/structure/transit_tube/Initialize(mapload, loc, newdirection)
```
Comment (Cyberboss) [inline code/game/objects/structures/transit_tubes/transit_tube.dm:16]: ^

File: code/game/objects/structures/transit_tubes/transit_tube.dm [hunk 6]
```diff
@@ -13,8 +13,8 @@
...
+	. = ..(mapload, loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/transit_tubes/transit_tube.dm:17]: just use ..() to KISS

File: code/modules/admin/verbs/adminhelp.dm [hunk 7]
```diff
@@ -138,7 +138,7 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 /obj/effect/statclick/ticket_list
 	var/current_state
 
-/obj/effect/statclick/ticket_list/New(loc, name, state)
+/obj/effect/statclick/ticket_list/Initialize(mapload, loc, name, state)
```
Comment (Cyberboss) [inline code/modules/admin/verbs/adminhelp.dm:141]: loc not a param

File: code/modules/antagonists/cult/blood_magic.dm [hunk 8]
```diff
@@ -358,11 +358,11 @@
 	var/health_cost = 0 //The amount of health taken from the user when invoking the spell
 	var/datum/action/innate/cult/blood_spell/source
 
-/obj/item/melee/blood_magic/New(loc, spell)
+/obj/item/melee/blood_magic/Initialize(maplod, loc, spell)
 	source = spell
 	uses = source.charges
 	health_cost = source.health_cost
-	..()
+	. = ..()
```
Comment (Cyberboss) [inline code/modules/antagonists/cult/blood_magic.dm:365]: loc not a param

File: code/modules/antagonists/wizard/equipment/artefact.dm [hunk 9]
```diff
@@ -44,13 +44,19 @@
 	var/spawn_amt_left = 20
 	var/spawn_fast = FALSE
 
-/obj/effect/rend/New(loc, spawn_type, spawn_amt, desc, spawn_fast)
+/obj/effect/rend/Initialize(mapload, loc, spawn_type, spawn_amt, desc, spawn_fast)
```
Comment (Cyberboss) [inline code/modules/antagonists/wizard/equipment/artefact.dm:47]: loc not a param

File: code/modules/atmospherics/gasmixtures/gas_types.dm [hunk 10]
```diff
@@ -262,7 +262,7 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
 	appearance_flags = TILE_BOUND
 	vis_flags = NONE
 
-/obj/effect/overlay/gas/New(state, alph)
+/obj/effect/overlay/gas/Initialize(mapload, state, alph)
```
Comment (Cyberboss) [inline code/modules/atmospherics/gasmixtures/gas_types.dm:265]: first arg replaced

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 11]
```diff
@@ -212,8 +212,8 @@
 	var/list/air_vent_info = list()
 	var/list/air_scrub_info = list()
 
-/obj/machinery/airalarm/New(loc, ndir, nbuild)
-	..()
+/obj/machinery/airalarm/Initialize(mapload, loc, ndir, nbuild)
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/airalarm.dm:215]: loc not a param

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 12]
```diff
@@ -77,15 +77,15 @@
 		if(HAS_TRAIT(L, TRAIT_VENTCRAWLER_NUDE) || HAS_TRAIT(L, TRAIT_VENTCRAWLER_ALWAYS))
 			. += "<span class='notice'>Alt-click to crawl through it.</span>"
 
-/obj/machinery/atmospherics/New(loc, process = TRUE, setdir, init_dir = ALL_CARDINALS)
+/obj/machinery/atmospherics/Initialize(mapload, loc, process = TRUE, setdir, init_dir = ALL_CARDINALS)
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:80]: loc not a param

File: code/modules/atmospherics/machinery/bluespace_vendor.dm [hunk 13]
```diff
@@ -67,7 +67,7 @@
 	dir = EAST
 	pixel_x = -30
 
-/obj/machinery/bluespace_vendor/New(loc, ndir, nbuild)
+/obj/machinery/bluespace_vendor/Initialize(mapload, loc, ndir, nbuild)
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/bluespace_vendor.dm:70]: loc not a param

File: code/modules/buildmode/buttons.dm [hunk 14]
```diff
@@ -4,7 +4,7 @@
 	// If we don't do this, we get occluded by item action buttons
 	plane = ABOVE_HUD_PLANE
 
-/atom/movable/screen/buildmode/New(bld)
+/atom/movable/screen/buildmode/Initialize(mapload, bld)
```
Comment (Cyberboss) [inline code/modules/buildmode/buttons.dm:7]: first arg replaced

File: code/modules/buildmode/buttons.dm [hunk 15]
```diff
@@ -58,7 +58,7 @@
 /atom/movable/screen/buildmode/modeswitch
 	var/datum/buildmode_mode/modetype
 
-/atom/movable/screen/buildmode/modeswitch/New(bld, mt)
+/atom/movable/screen/buildmode/modeswitch/Initialize(mapload, bld, mt)
```
Comment (Cyberboss) [inline code/modules/buildmode/buttons.dm:61]: ^

File: code/modules/buildmode/buttons.dm [hunk 16]
```diff
@@ -72,7 +72,7 @@
 /atom/movable/screen/buildmode/dirswitch
 	icon_state = "build"
 
-/atom/movable/screen/buildmode/dirswitch/New(bld, dir)
+/atom/movable/screen/buildmode/dirswitch/Initialize(mapload, bld, dir)
```
Comment (Cyberboss) [inline code/modules/buildmode/buttons.dm:75]: ^

File: code/modules/buildmode/effects/line.dm [hunk 17]
```diff
@@ -2,7 +2,9 @@
 	var/image/I
 	var/client/cl
 
-/obj/effect/buildmode_line/New(client/C, atom/atom_a, atom/atom_b, linename)
+/obj/effect/buildmode_line/Initialize(mapload, client/C, atom/atom_a, atom/atom_b, linename)
```
Comment (Cyberboss) [inline code/modules/buildmode/effects/line.dm:5]: first arg replaced

File: code/modules/cargo/order.dm [hunk 18]
```diff
@@ -3,8 +3,8 @@
 	var/order_id = 0
 	var/errors = 0
 
-/obj/item/paper/fluff/jobs/cargo/manifest/New(atom/A, id, cost)
-	..()
+/obj/item/paper/fluff/jobs/cargo/manifest/Initialize(mapload, atom/A, id, cost)
+	. = ..()
```
Comment (Cyberboss) [inline code/modules/cargo/order.dm:7]: ^

File: code/modules/events/immovable_rod.dm [hunk 19]
```diff
@@ -73,7 +73,7 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 	/// Whether the rod can loop across other z-levels. The rod will still loop when the z-level is self-looping even if this is FALSE.
 	var/loopy_rod = FALSE
 
-/obj/effect/immovablerod/New(atom/start, atom/end, aimed_at, force_looping)
+/obj/effect/immovablerod/Initialize(mapload, atom/start, atom/end, aimed_at, force_looping)
```
Comment (Cyberboss) [inline code/modules/events/immovable_rod.dm:76]: ^

File: code/modules/mob/living/simple_animal/guardian/types/support.dm [hunk 20]
```diff
@@ -98,7 +98,7 @@
 	anchored = TRUE
 	layer = ABOVE_OPEN_TURF_LAYER
 
-/obj/structure/receiving_pad/New(loc, mob/living/simple_animal/hostile/guardian/healer/G)
+/obj/structure/receiving_pad/Initialize(mapload, loc, mob/living/simple_animal/hostile/guardian/healer/G)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/guardian/types/support.dm:101]: loc not a param

File: code/modules/modular_computers/computers/item/processor.dm [hunk 21]
```diff
@@ -19,8 +19,8 @@
 	machinery_computer = null
 	. = ..()
 
-/obj/item/modular_computer/processor/New(comp)
-	..()
+/obj/item/modular_computer/processor/Initialize(mapload, comp)
```
Comment (Cyberboss) [inline code/modules/modular_computers/computers/item/processor.dm:22]: first arg replaced

File: code/modules/paperwork/contract.dm [hunk 22]
```diff
@@ -17,11 +17,10 @@
 /obj/item/paper/contract/employment
 	icon_state = "paper_words"
 
-/obj/item/paper/contract/employment/New(atom/loc, mob/living/nOwner)
+/obj/item/paper/contract/employment/Initialize(mapload, atom/loc, mob/living/nOwner)
```
Comment (Cyberboss) [inline code/modules/paperwork/contract.dm:20]: loc not a param

File: code/modules/power/apc.dm [hunk 23]
```diff
@@ -207,12 +207,12 @@
 	if(terminal)
 		terminal.connect_to_network()
 
-/obj/machinery/power/apc/New(turf/loc, ndir, building=0)
+/obj/machinery/power/apc/Initialize(mapload, turf/loc, ndir, building=0)
```
Comment (Cyberboss) [inline code/modules/power/apc.dm:210]: loc not a param

File: code/modules/projectiles/pins.dm [hunk 24]
```diff
@@ -15,8 +15,8 @@
 	var/pin_removeable = FALSE // Can be replaced by any pin.
 	var/obj/item/gun/gun
 
-/obj/item/firing_pin/New(newloc)
-	..()
+/obj/item/firing_pin/Initialize(mapload, newloc)
```
Comment (Cyberboss) [inline code/modules/projectiles/pins.dm:18]: loc not a param

File: code/modules/spells/spell_types/personality_commune.dm [hunk 25]
```diff
@@ -10,7 +10,7 @@
 	var/datum/brain_trauma/severe/split_personality/trauma
 	var/flufftext = "You hear an echoing voice in the back of your head..."
 
-/obj/effect/proc_holder/spell/targeted/personality_commune/New(datum/brain_trauma/severe/split_personality/T)
+/obj/effect/proc_holder/spell/targeted/personality_commune/Initialize(mapload, datum/brain_trauma/severe/split_personality/T)
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/personality_commune.dm:13]: first arg replaced

File: code/modules/surgery/organs/augments_internal.dm [hunk 26]
```diff
@@ -8,7 +8,7 @@
 	var/implant_overlay
 	var/syndicate_implant = FALSE //Makes the implant invisible to health analyzers and medical HUDs.
 
-/obj/item/organ/cyberimp/New(mob/implanted_mob = null)
+/obj/item/organ/cyberimp/Initialize(mapload, mob/implanted_mob = null)
```
Comment (Cyberboss) [inline code/modules/surgery/organs/augments_internal.dm:11]: first arg replaced

File: code/datums/brain_damage/special.dm [hunk 27]
```diff
@@ -378,10 +378,10 @@
 	image_state = "secbot-c"
 	var/victim
 
-/obj/effect/hallucination/simple/securitron/New()
+/obj/effect/hallucination/simple/securitron/Initialize()
 	name = pick ( "officer Beepsky", "officer Johnson", "officer Pingsky")
 	START_PROCESSING(SSfastprocess,src)
-	..()
+	. = ..()
```
Comment (Fikou) [inline code/datums/brain_damage/special.dm:384]: do return ..() at the end of procs

File: code/game/objects/items/implants/implant_track.dm [hunk 28]
```diff
@@ -15,13 +15,13 @@
...
-/obj/item/implant/tracking/New()
-	..()
+/obj/item/implant/tracking/Initialize()
+	. = ..()
 	GLOB.tracked_implants += src
 
 /obj/item/implant/tracking/Destroy()
-	. = ..()
 	GLOB.tracked_implants -= src
+	. = ..()
```
Comment (Fikou) [inline code/game/objects/items/implants/implant_track.dm:24]: same here

File: code/game/turfs/closed/_closed.dm [hunk 29]
```diff
@@ -60,11 +60,11 @@
 	plane = SPLASHSCREEN_PLANE
 	bullet_bounce_sound = null
 
-/turf/closed/indestructible/splashscreen/New()
+/turf/closed/indestructible/splashscreen/Initialize()
 	SStitle.splash_turf = src
 	if(SStitle.icon)
 		icon = SStitle.icon
-	..()
+	. = ..()
```
Comment (Fikou) [inline code/game/turfs/closed/_closed.dm:67]: and here

File: code/modules/antagonists/cult/blood_magic.dm (hunk 8 above)
Comment (Fikou) [reply code/modules/antagonists/cult/blood_magic.dm:365]: and here

File: code/modules/ruins/objects_and_mobs/sin_ruins.dm [hunk 30]
```diff
@@ -111,12 +111,13 @@
...
 
-/obj/structure/mirror/magic/pride/New()
+/obj/structure/mirror/magic/pride/Initialize()
 	for(var/speciestype in subtypesof(/datum/species))
 		var/datum/species/S = speciestype
 		if(initial(S.changesource_flags) & MIRROR_PRIDE)
 			choosable_races += initial(S.id)
-	..()
+	// Populate choosable_races before calling super, to supress default behaviour
+	. = ..()
```
Comment (Fikou) [inline code/modules/ruins/objects_and_mobs/sin_ruins.dm:120]: n here

File: code/datums/brain_damage/special.dm (hunk 27 above)
Comment (Fikou) [reply code/datums/brain_damage/special.dm:384]: should this even be in this order?

Comment (Mothblocks) [reply code/datums/brain_damage/special.dm:384]: no, `. = ..()` should be first

Comment (LemonInTheDark) [reply code/datums/brain_damage/special.dm:384]: first in init, last in destroy

File: code/game/objects/structures/mirror.dm [hunk 31]
```diff
@@ -139,25 +139,25 @@
 	icon_state = "magic_mirror"
 	var/list/choosable_races = list()
 
-/obj/structure/mirror/magic/New()
+/obj/structure/mirror/magic/Initialize()
+	. = ..()
```
Comment (Rohesie) [inline code/game/objects/structures/mirror.dm:143]: Do something like this here:
```dm
/obj/structure/mirror/magic/Initialize()
	. = ..()
	setup_choosable_races()

/obj/structure/mirror/magic/proc/setup_choosable_races()
	if(!choosable_races.len)
		(...)
```

That way you can call parent first on Initialize and redefine that proc for `/obj/structure/mirror/magic/lesser` and the rest with their own logic that may come before.

Comment (coiax) [reply code/game/objects/structures/mirror.dm:143]: Very smart.

**Comment by coiax** [comment]
Will split into smaller PRs.

**Pull request closed without merging**